src/gpu/nvidia/sycl_cuda_memory_storage_helper.hpp

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+* Copyright 2022 Intel Corporation
+* Copyright 2022 Codeplay Software Limited
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_NVIDIA_SYCL_CUDA_MEMORY_STORAGE_HELPER_HPP
+#define GPU_NVIDIA_SYCL_CUDA_MEMORY_STORAGE_HELPER_HPP
+
+#include <optional>
+#include "gpu/nvidia/sycl_cuda_scoped_context.hpp"
+#include "sycl/sycl_memory_storage.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace nvidia {
+
+#define CTX_IN_SYCL_MEMORY(arg) \
+    sycl_memory_arg_t<::sycl::access::mode::read>(&CTX_IN_STORAGE(arg), cgh)
+
+#define CTX_OUT_SYCL_MEMORY(arg) \
+    sycl_memory_arg_t<::sycl::access::mode::write>(&CTX_OUT_STORAGE(arg), cgh)
+
+#define CTX_SCRATCH_SYCL_MEMORY(arg) \
+    sycl_memory_arg_t<::sycl::access::mode::read_write>( \
+            ctx.get_scratchpad_grantor().get_memory_storage(arg).get(), cgh)
+
+template <::sycl::access_mode mode>
+class sycl_memory_arg_t {
+public:
+    sycl_memory_arg_t() = default;
+    sycl_memory_arg_t(memory_storage_t *raw_mem, ::sycl::handler &cgh) {
+        if (raw_mem->is_null()) { return; }
```

Comment:
Let's extend the check to `(!raw_mem || raw_mem->is_null())` so that `CTX_SCRATCH_SYCL_MEMORY` can be used unconditionally.

Reply:
done